crates/router/src/types/storage/payment_method.rs

```diff
@@ -81,7 +81,7 @@ impl PaymentTokenData {
 #[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub struct PaymentMethodListContext {
     pub card_details: Option<api::CardDetailFromLocker>,
-    pub hyperswitch_token_data: PaymentTokenData,
+    pub hyperswitch_token_data: Option<PaymentTokenData>,
```

Comment:
Why we are moving to Option<PaymentTokenData>? when `PaymentMethodListContext` don't have payment token?

Reply:
This change comes from https://github.com/juspay/hyperswitch/pull/4856
It reflects in this PR as it builds upon that one.

There are two flows for Saved payment list,
- SDK based with payment id, where this token would be present
- S2S call for without payment's context where payment related info would not generated.